src/modals/SendModal/index.tsx

```diff
@@ -59,40 +74,59 @@ const SendModal = ({ onClose }: SendModalProps) => {
     currentNetwork
   } = useGlobalContext()
   const { setAddress } = useAddressesContext()
+  const { dappTransactionData, requestEvent, walletConnect } = useWalletConnectContext()
   const [title, setTitle] = useState('')
-  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>()
+  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>(dappTransactionData)
   const [isLoading, setIsLoading] = useState(false)
-  const [step, setStep] = useState<StepIndex>(1)
+  const [step, setStep] = useState<StepIndex>(StepIndex.Send)
   const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
   const [consolidationRequired, setConsolidationRequired] = useState(false)
   const [isSweeping, setIsSweeping] = useState(false)
   const [sweepUnsignedTxs, setSweepUnsignedTxs] = useState<SweepAddressTransaction[]>([])
   const [unsignedTxId, setUnsignedTxId] = useState('')
   const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [contractAddress, setContractAddress] = useState('')
   const [fees, setFees] = useState<bigint>()
   const theme = useTheme()
 
   useEffect(() => {
-    if (step === 1) {
-      setTitle('Send')
-    } else if (step === 2) {
-      setTitle('Info Check')
-    } else if (step === 3) {
-      setTitle('Password Check')
-    }
+    setTitle(stepToTitle.get(step) || 'Unknown')
   }, [setStep, setTitle, step])
 
   const confirmPassword = () => {
     if (consolidationRequired) setIsConsolidateUTXOsModalVisible(false)
-    setStep(3)
+    setStep(StepIndex.PasswordCheck)
   }
 
   const buildTransaction = async (transactionData: SendTransactionData) => {
-    setTransactionData(transactionData)
+    const { fromAddress, toAddress, gasAmount, gasPrice, contractCode, contractState, issueTokenAmount, script } =
+      transactionData
+    let { amount } = transactionData
+
+    const isContractTx = contractCode !== ''
+    const isScriptTx = script !== ''
+
+    if (isContractTx) {
+      if (transactionData.amount == '') {
+        amount = '0.000001'
```

Comment:
Maybe we should extract this value as a constant somewhere?

Reply:
Yeah I'd like that... to `utils/constants`!